src/build_workflow/builder_from_dist.py

```diff
@@ -40,14 +41,28 @@ def export_artifacts(self, build_recorder):
             logging.info(f"Downloading into {artifact_path} ...")
             if artifact_type not in ["maven"]:  # avoid re-publishing maven artifacts, see https://github.com/opensearch-project/opensearch-build/issues/1279
                 for artifact in component_manifest.artifacts[artifact_type]:
-                    artifact_url = f"{self.component.dist}/{self.target.platform}/{self.target.architecture}/builds/{self.target_name}/{artifact}"
-                    artifact_dest = os.path.realpath(os.path.join(self.output_path, artifact))
-                    os.makedirs(os.path.dirname(artifact_dest), exist_ok=True)
-                    logging.info(f"Downloading {artifact_url} into {artifact_dest}")
-                    urllib.request.urlretrieve(artifact_url, artifact_dest)
-                    build_recorder.record_artifact(self.component.name, artifact_type, artifact, artifact_dest)
+                    for url_pattern in self.__get_url_patterns():
+                        artifact_url = f"{url_pattern}/{artifact}"
+                        try:
```

Comment:
This swallows too many exceptions and fails silently if a distribution cannot be found. 

1. Add a `distribution_urls` method that returns the list of possible URLs.
2. Add a `distribution_url` method that iterates over `distribution_urls` and returns the first one that has a manifest.yml
3. Use the result from `distribution_url` everywhere.